src/models/event.ts

```diff
@@ -579,9 +579,18 @@ export class MatrixEvent extends TypedEventEmitter<MatrixEventEmittedEvents, Mat
         const relatesTo = this.getWireContent()?.["m.relates_to"];
         if (relatesTo?.rel_type === THREAD_RELATION_TYPE.name) {
             return relatesTo.event_id;
-        } else {
-            return this.getThread()?.id || this.threadId;
         }
+        if (this.thread) {
+            return this.thread.id;
+        }
+        if (this.threadId != undefined) {
+            return this.threadId;
+        }
```

Comment:
Probably not worth doing as part of this pull request, but `this.threadId` has a lot of overlap with the `unsigned.thread_id` field.
We should probably factor it out, and consolidate it under what MSC4023 proposes.